library/src/main/java/com/feedhenry/sdk/android/SyncableActivity.java

```diff
@@ -13,26 +13,18 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package com.feedhenry.sdk.api;
+package com.feedhenry.sdk.android;
 
-import android.support.test.runner.AndroidJUnit4;
+import com.feedhenry.sdk.sync.FHSyncListener;
 
-import com.feedhenry.sdk.Device;
-
-import org.junit.Test;
-import org.junit.runner.RunWith;
-
-import static android.support.test.InstrumentationRegistry.getContext;
-import static junit.framework.Assert.assertNotNull;
-
-@RunWith(AndroidJUnit4.class)
-public class DeviceTest {
-
-    @Test
-    public void testDevice() {
-        assertNotNull(Device.getDeviceId(getContext()));
-        assertNotNull(Device.getDeviceModelAndManufacturer());
-        assertNotNull(Device.getUserAgent());
-    }
+/**
+ * Implement this for activities that use sync.
+ */
+public interface SyncableActivity {
```

Comment:
I am not really sure if this is a good idea. There are a few problems:

1. The way how `onBindSyncListener` being invoked does feel right to me. If I am not mistaken, it will not work when there are 2 instances of `SyncActivity` being created to display 2 different datasets. See my comments above. 
2. I think this approach is against `MV**` patterns.  If you think about it, the sync framework is mainly for fetching data,  so it is mainly work with `Models`.  On the other hand, however, the main job of `Activity` are for `Views`. In many applications, when developers apply `MVC` or `MVP` in their applications, they will have their own abstractions for different roles. More specifically, there will be `Controllers/Presenters/ViewModels` that are responsible for the communications between `Models` and `Views`.

But the introduction of this activity forces developers to make `Views` (the implementations of this activity) to communicate with the `Models` (the data that are managed by the Sync framework) directly. It also makes the Sync framework a hard dependency of this activity, making it hard to unit test this activity.

I think it's better not to provide this interface, but rather provider an example to show how to wrap the Sync framework as a `Repository`. Developers can then inject the Repository into there controller/presenters, which will control how the data will be fetched, and how to pass them to the views.

Reply:
I don't like it too. It was based on simple use-case in sync-android-app. Yes, now I see that it can lead developers to bad practice. 
User can still handle this manually as it was.

Reply:
Let's hack on some ideas around this.  @wei-lee is on the right track.

Even if we don't use it in sync, RxJava has a LOT of concepts which are pretty much  our use cases.  I say let's make a reading list to get some better ideas.